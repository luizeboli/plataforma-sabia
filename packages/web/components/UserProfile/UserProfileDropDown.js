import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import LogoutButton from './LogoutButton';
import PageLink from './PageLink';
import getPages from './pages';

const UserProfileDropDown = ({ visible, toggleVisible }) => {
	const { t } = useTranslation(['profile']);

	return (
		visible && (
			<DropDownContainer>
				<DropDownMenu>
					{getPages(t).map(({ pages }) =>
						pages.map((page) => (
							<li key={page.title}>
								<PageLink href={page.href} onClick={toggleVisible}>
									<page.icon />
									{page.title}
								</PageLink>
							</li>
						)),
					)}
					<li>
						<LogoutButton cb={toggleVisible} />
					</li>
				</DropDownMenu>
			</DropDownContainer>
		)
	);
};

UserProfileDropDown.propTypes = {
	visible: PropTypes.bool.isRequired,
	toggleVisible: PropTypes.func.isRequired,
};

const DropDownContainer = styled.div`
	position: relative;
`;

const DropDownMenu = styled.ul`
	${({ theme: { colors, metrics } }) => css`
		position: absolute;
		width: 22rem;
		left: calc(50% - 11rem);
		top: calc(100% + 2.1rem);
		background: ${colors.white};
		border-radius: ${metrics.baseRadius}rem;
		padding: 2rem 1.5rem;
		box-shadow: 0 0 2rem -1.5rem ${colors.secondary};
		transition: 0.3s;

		:hover {
			box-shadow: 0 0 2.2rem -1.5rem ${colors.secondary};
		}

		&::before {
			content: '';
			position: absolute;
			left: calc(50% - 2rem);
			top: -2rem;
			width: 0;
			height: 0;
			border-left: 2rem solid transparent;
			border-right: 2rem solid transparent;
			border-bottom: 2rem solid ${colors.white};
		}

		li:last-child {
			padding-top: 1rem;
			border-top: 0.1rem solid ${colors.border};
		}
	`}
`;

export default UserProfileDropDown;
