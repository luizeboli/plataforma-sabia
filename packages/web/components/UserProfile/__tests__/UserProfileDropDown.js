import React from 'react';
import { render, screen } from 'test-utils';
import UserProfileDropDown from '../UserProfileDropDown';

describe('UserProfileDropDown', () => {
	test('it renders as expected if visible is true', () => {
		const { container } = render(<UserProfileDropDown visible toggleVisible={() => {}} />);

		expect(screen.queryAllByRole('list').length).toBeGreaterThan(0);
		expect(screen.queryByRole('button', { name: /sair|logout/i })).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	test('it does not render if receives visible as false', () => {
		const { container } = render(
			<UserProfileDropDown visible={false} toggleVisible={() => {}} />,
		);

		expect(screen.queryAllByRole('list')).toHaveLength(0);
		expect(screen.queryByRole('button', { name: /sair|logout/i })).toBeNull();
		expect(container).toMatchSnapshot();
	});
});
