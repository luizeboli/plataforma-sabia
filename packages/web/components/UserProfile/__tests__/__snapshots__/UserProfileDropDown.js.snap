// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileDropDown it does not render if receives visible as false 1`] = `<div />`;

exports[`UserProfileDropDown it renders as expected if visible is true 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1.6rem;
  background: none;
  border: none;
  color: hsla(0,0%,40%);
}

.c3:hover {
  color: hsla(146,64%,36%);
}

.c3 svg {
  fill: hsla(0,0%,40%);
  stroke: hsla(0,0%,40%);
  width: 2rem;
  height: 2rem;
  margin-right: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.6rem;
  margin-bottom: 2rem;
  color: hsla(0,0%,40%);
}

.c2:hover {
  color: hsla(146,64%,36%);
}

.c2 svg {
  fill: hsla(0,0%,40%);
  stroke: hsla(0,0%,40%);
  width: 2rem;
  height: 2rem;
  margin-right: 1rem;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  width: 22rem;
  left: calc(50% - 11rem);
  top: calc(100% + 2.1rem);
  background: hsla(0,0%,100%);
  border-radius: 0.5rem;
  padding: 2rem 1.5rem;
  box-shadow: 0 0 2rem -1.5rem hsl(166,100%,34%);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c1:hover {
  box-shadow: 0 0 2.2rem -1.5rem hsl(166,100%,34%);
}

.c1::before {
  content: '';
  position: absolute;
  left: calc(50% - 2rem);
  top: -2rem;
  width: 0;
  height: 0;
  border-left: 2rem solid transparent;
  border-right: 2rem solid transparent;
  border-bottom: 2rem solid hsla(0,0%,100%);
}

.c1 li:last-child {
  padding-top: 1rem;
  border-top: 0.1rem solid hsla(0,0%,93%);
}

<div>
  <div
    class="c0"
  >
    <ul
      class="c1"
    >
      <li>
        <a
          class="c2"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
            />
          </svg>
          profile:myProfile
        </a>
      </li>
      <li>
        <a
          class="c2"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm-6 400H54a6 6 0 0 1-6-6V86a6 6 0 0 1 6-6h404a6 6 0 0 1 6 6v340a6 6 0 0 1-6 6zm-42-92v24c0 6.627-5.373 12-12 12H204c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h200c6.627 0 12 5.373 12 12zm0-96v24c0 6.627-5.373 12-12 12H204c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h200c6.627 0 12 5.373 12 12zm0-96v24c0 6.627-5.373 12-12 12H204c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h200c6.627 0 12 5.373 12 12zm-252 12c0 19.882-16.118 36-36 36s-36-16.118-36-36 16.118-36 36-36 36 16.118 36 36zm0 96c0 19.882-16.118 36-36 36s-36-16.118-36-36 16.118-36 36-36 36 16.118 36 36zm0 96c0 19.882-16.118 36-36 36s-36-16.118-36-36 16.118-36 36-36 36 16.118 36 36z"
            />
          </svg>
          profile:myTechnologies
        </a>
      </li>
      <li>
        <a
          class="c2"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128 480h256V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v400zm64-384h128v32H192V96zm320 80v256c0 26.5-21.5 48-48 48h-48V128h48c26.5 0 48 21.5 48 48zM96 480H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h48v352z"
            />
          </svg>
          profile:myNegotiations
        </a>
      </li>
      <li>
        <a
          class="c2"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            />
          </svg>
          profile:messages
        </a>
      </li>
      <li>
        <a
          class="c2"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          profile:favoriteTechnologies
        </a>
      </li>
      <li>
        <button
          class="c3"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 0 1-112.7 75.9A352.8 352.8 0 0 1 512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 0 1-112.7-75.9 353.28 353.28 0 0 1-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 0 0 0-12.6z"
            />
          </svg>
          logout
        </button>
      </li>
    </ul>
  </div>
</div>
`;
